### PR TITLE
[DMS-68] Move case on direction to the outside of the main iteration function

### PR DESCRIPTION
**Pull request by s-and-witch**
| |size|foldLeft|foldRight|mapfilter|map|
|--:|--:|--:|--:|--:|--:|
|persistentmap|100|79_472|79_588|157_392|29_048|
|persistentmap_baseline|100|92_138|93_745|169_663|29_048|
|persistentmap|1000|761_667|762_619|3_430_215|257_766|
|persistentmap_baseline|1000|888_169|899_681|3_556_717|257_766|
|persistentmap|10000|7_582_712|18_270_414|42_053_473|2_544_359|
|persistentmap_baseline|10000|19_529_035|19_640_763|43_314_564|2_544_359|
|persistentmap|100000|182_610_151|182_615_427|493_211_373|132_210_500|
|persistentmap_baseline|100000|195_212_923|196_318_318|505_815_170|132_210_500|
|persistentmap|1000000|1_826_050_546|1_826_067_207|5_637_839_066|1_322_035_748|
|persistentmap_baseline|1000000|1_952_082_879|1_963_098_142|5_763_869_513|1_322_035_748|

### CODE REVIEW

File: src/PersistentOrderedMap.mo [hunk 1]
```diff
@@ -324,27 +324,49 @@ module {
...
+        case (?(#tr(#leaf), ts)) {
+          trees := ts;
+          next()
+        };
+        case (?(#xy(xy), ts)) {
+          trees := ts;
+          ?xy
+        };
+        case (?(#tr(#node(_, l, xy, r)), ts)) {
+          trees := ?(#tr(l), ?(#xy(xy), ?(#tr(r), ts)));
```
Comment (GoPavel) [inline src/PersistentOrderedMap.mo:347]: I see that this function is different only in the order in which we are putting elements on the stack.
Can we somehow avoid copy-paste via small functions `turnLeftFirst`/`turnRightFirst` or something like that and hope they will be inlined?

Comment (GoPavel) [reply src/PersistentOrderedMap.mo:347]: I mean to have something like:
```motoko
public func iter<K, V>(rbMap : Map<K, V>, direction : Direction) : I.Iter<(K, V)> {
    switch direction {
      case (#fwd) { iter(rbMap, turnLeftFirst) };
      case (#bwd) { iter(rbMap, turnRightFirst) }
    }
  };
```

Comment (s-and-witch) [reply src/PersistentOrderedMap.mo:347]: See #17

**Comment by GoPavel** [comment]
Note: do not merge since #17 is favored due to more clean code

**Pull request closed without merging**